test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/persistence/WorkbenchPersistenceIntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.api.model.ServiceResponse.ResponseType;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.integrationtests.controller.client.KieServerMgmtControllerClient;
+
+public class WorkbenchPersistenceIntegrationTest {
+
+    private static final String ORGANIZATION_UNIT_NAME = "myOrgUnit";
+    private static final String REPOSITORY_NAME = "myRepo";
+
+    private static final String PROJECT_GROUP_ID = "org.kie.server.testing";
+    private static final String PROJECT_NAME = "definition-project";
+    private static final String PROJECT_VERSION = "1.0.0.Final";
+
+    private static final String CONTAINER_ID = "cont-id";
+    private static final String CONTAINER_ALIAS = "cont-alias";
+
+    private static DeploymentScenarioBuilderFactory deploymentScenarioFactory = DeploymentScenarioBuilderFactoryLoader.getInstance();
+    private WorkbenchWithKieServerScenario workbenchWithKieServerScenario;
+
+    private WorkbenchClientProvider workbenchClientProvider;
+    private KieServerMgmtControllerClient kieControllerClient;
+    private KieServerClientProvider kieServerClientProvider;
+    private GitProvider gitProvider;
+
+    @Before
+    public void setUp() {
+        workbenchWithKieServerScenario = deploymentScenarioFactory.getWorkbenchWithKieServerScenarioBuilder().build();
+        workbenchWithKieServerScenario.deploy();
+
+        gitProvider = GitProviderFactory.getGitProvider();
+        gitProvider.createGitRepository(workbenchWithKieServerScenario.getNamespace(), ClassLoader.class.getResource("/kjars-sources").getFile());
+
+        workbenchClientProvider = new WorkbenchClientProvider(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        kieServerClientProvider = new KieServerClientProvider(workbenchWithKieServerScenario.getKieServerDeployment());
+        kieControllerClient = new KieServerMgmtControllerClient(workbenchWithKieServerScenario.getWorkbenchDeployment().getUrl().toString() + "/rest/controller",
+                workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername(), workbenchWithKieServerScenario.getWorkbenchDeployment().getPassword());
+    }
+
+    @After
+    public void tearDown() {
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getKieServerDeployment());
+        workbenchWithKieServerScenario.undeploy();
+        gitProvider.deleteGitRepository(workbenchWithKieServerScenario.getNamespace());
+    }
+
+    @Test
+    public void testWorkbenchControllerPersistence() {
+        workbenchClientProvider.createOrganizationalUnit(ORGANIZATION_UNIT_NAME, workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername());
```

Comment:
Confusing naming. I'd expect WorkbenchClientProvider to provide some WorkbenchClient. But here it's being used as the client itself. How about renaming WorkbenchClientProvider as WorkbenchClient or similar (and simplify variable name as well)?